netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -1803,9 +1906,24 @@ net_ebpf_extension_sock_addr_redirect_connection_classify(
     }
 
     // Populate the sock_addr context with WFP classify input fields.
-    _net_ebpf_extension_sock_addr_copy_wfp_connection_fields(
-        incoming_fixed_values, incoming_metadata_values, &net_ebpf_sock_addr_ctx);
+    // If the filter context is wildcard filter context, use layer data to populate context.
+    if (filter_context->base.wildcard) {
+        _net_ebpf_extension_sock_addr_copy_wfp_connection_fields(
+            incoming_fixed_values, incoming_metadata_values, NULL, &net_ebpf_sock_addr_ctx);
+    } else {
+        _net_ebpf_extension_sock_addr_copy_wfp_connection_fields(
+            incoming_fixed_values, incoming_metadata_values, NULL, &net_ebpf_sock_addr_ctx);
+    }
+
+    if (net_ebpf_sock_addr_ctx.flags & FWP_CONDITION_FLAG_IS_REAUTHORIZE) {
+        // In case of re-authorization, the verdict is always to proceed (terminating).
```

Comment:
please elaborate that re-auth means that the eBPF programs have already inspected the connection and we will skip invoking the programs. Line 1920 is redundant and may cause confusion. I suggest you remove it.

Reply:
by line 1920 do you mean call to `FwpsQueryConnectionRedirectState()`? Will that not be needed for the case where, after redirection, the proxy creates a new connection to the destination, and we want to skip invoking program for that new connection?